File: examples/rag_demo_grpc.js

const { GrpcClient } = require('../barq-sdk-ts/dist/index');
const path = require('path');

async function main() {
    // Path to proto file (assuming running from project root)
    const protoPath = path.join(__dirname, '../barq-sdk-ts/proto/barq.proto');

    console.log("Connecting to Barq gRPC at localhost:50051...");
    const client = new GrpcClient('localhost:50051', protoPath);

    try {
        const health = await client.health();
        console.log("Health check:", health);

        console.log("Creating collection 'grpc_ts_rag'...");
        await client.createCollection('grpc_ts_rag', 2, 'Cosine');

        console.log("Inserting document...");
        await client.insertDocument('grpc_ts_rag', 'doc_ts_1', [0.9, 0.1], { lang: 'ts', type: 'grpc' });

        // Wait for consistency
        await new Promise(r => setTimeout(r, 500));

        console.log("Searching...");
        const results = await client.search('grpc_ts_rag', [0.9, 0.1], 3);
        console.log("Found results:");
        console.log(results);

    } catch (e) {
        console.error("Error:", e);
    }
}

main();

File: barq-sdk-rust/examples/rag_demo_grpc.rs
use barq_sdk_rust::{BarqGrpcClient, DistanceMetric, DocumentId};
use serde_json::json;
use tokio::time::Duration;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // 1. Connect to Barq using gRPC (port 50051)
    let mut client = BarqGrpcClient::connect("http://localhost:50051".to_string()).await?;
    println!("Connected to Barq gRPC at localhost:50051");

    // 2. Health check
    let health = client.health().await?;
    println!("Health check: {}", health);

    // 3. Create Collection
    println!("Creating collection 'grpc_rag_demo'...");
    client.create_collection("grpc_rag_demo", 4, DistanceMetric::Cosine).await?;

    // 4. Insert Document
    println!("Inserting documents...");
    let doc_id = DocumentId::Str("doc_grpc_1".to_string());
    let vector = vec![0.1, 0.2, 0.3, 0.4];
    let payload = json!({"text": "Hello form gRPC", "category": "demo"});
    
    client.insert_document("grpc_rag_demo", doc_id, vector.clone(), payload).await?;

    // Wait for indexing (simple consistency)
    tokio::time::sleep(Duration::from_millis(500)).await;

    // 5. Search
    println!("Searching...");
    let results = client.search("grpc_rag_demo", vector, 3).await?;
    println!("Found {} results:", results.len());
    for res in results {
        println!(" - {:?}", res);
    }

    Ok(())
}
